### ISSUE TITLE
Relative URLs not resolved correctly by RefUtils.buildUrl

### ISSUE THREAD

**Issue by jweisman** [swagger-api/swagger-parser#1035]
Relative links are not working in schema definitions. 

* Starting with a "/" results in an incorrectly built URL (i.e. the beginning slash is not taken into consideration in `buildUrl`. We need something like:
```
        if(relPathParts[0].equals("/")) {
            trimRoot = rootPathParts.length-3;
        }      
```
* Relative path traversal results in an `IndexOutOfBoundsException`, I believe because this line is incorrect:
```
rootPathParts.length - trimRoot + trimRel - 1
```
Looks like it should be:
```
rootPathParts.length - trimRoot - trimRel + 1
```

You can test with this definition:
https://s3-eu-west-1.amazonaws.com/weisman.me/openapi/petstore-3.0.yaml

Which links to `/weisman.me/openapi/category.yaml#/Category`

**Comment by gracekarina** [comment]
Closed by #1038

**Comment by sandeepveethu** [comment]
Are there any plans to make this fix available in swagger-parser 1.x ? I am using swagger-parser 1.0.53 and facing similar issue where `buildUrl(rootPath, relativePath)` method is building incorrect url.
